sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DaysWritable.scala

```diff
@@ -35,11 +35,12 @@ import org.apache.spark.sql.catalyst.util.DateTimeUtils.{rebaseGregorianToJulian
  * @param julianDays The number of days since the epoch 1970-01-01 in
  *                   Julian calendar.
  */
-private[hive] class DaysWritable(
+class DaysWritable(
     var gregorianDays: Int,
     var julianDays: Int)
   extends DateWritable {
 
+  def this() = this(0, 0)
```

Comment:
I assume that `gregorianDays` and `julianDays` will be set later via the `set` method.